src/IW/Core/WithId.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE DeriveAnyClass #-}
+
+module IW.Core.WithId 
+       ( WithId (..)
+       ) where
+
+import IW.Core.Id (Id (..))
+
+
+data WithId a = WithId
```

Comment:
Let's add some documentation on this data type is needed 🙂 
We will need to write some Haddocks at some point, so other people can understand our codebase. But this can be done lazily when we are working on some features.